modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -313,6 +313,136 @@ Error GDScriptAnalyzer::resolve_inheritance(GDScriptParser::ClassNode *p_class,
 	class_type.native_type = result.native_type;
 	p_class->set_datatype(class_type);
 
+	#define CHECK_MEMBER_AGAINST_INHERITANCE(name, node) \
+	{ \
+		if (current_class_node->members_indices.has(name)) { \
+			int index = current_class_node->members_indices[name]; \
+			const GDScriptParser::ClassNode::Member *base_member = &current_class_node->members[index]; \
+			\
+			if (base_member->type == GDScriptParser::ClassNode::Member::VARIABLE || \
+				base_member->type == GDScriptParser::ClassNode::Member::CONSTANT || \
+				base_member->type == GDScriptParser::ClassNode::Member::ENUM || \
+				base_member->type == GDScriptParser::ClassNode::Member::ENUM_VALUE || \
+				base_member->type == GDScriptParser::ClassNode::Member::CLASS) { \
+					push_error(vformat(R"(The member "%s" already exists in a parent class.)", name), node); \
+					return ERR_PARSE_ERROR; \
+			} else if (base_member->type == GDScriptParser::ClassNode::Member::SIGNAL && p_class->members[i].type == GDScriptParser::ClassNode::Member::SIGNAL) { \
+				push_error(vformat(R"(The signal "%s" already exists in a parent class.)", name), node); \
+				return ERR_PARSE_ERROR; \
+			} \
```

Comment:
I think signal should not clash with any of the others, since it's first class now. If a signal clashes with a method or variable it can have unintended effects.